iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileResource.java

```diff
@@ -388,12 +388,29 @@ public long getTsFileSize() {
   }
 
   public long getStartTime(IDeviceID deviceId) {
-    return deviceId == null ? getFileStartTime() : timeIndex.getStartTime(deviceId);
+    try {
+      return deviceId == null ? getFileStartTime() : timeIndex.getStartTime(deviceId);
+    } catch (Exception e) {
+      LOGGER.error(
+          "meet error when getStartTime of {} in file {}", deviceId, file.getAbsolutePath());
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("TimeIndex = {}", timeIndex.toString());
```

Comment:
Unnecessary toString() and thus if-clause.
How about also reporting the exception?

Reply:
Fixed